awx/main/tasks/system.py

```diff
@@ -485,13 +485,17 @@ def inspect_execution_nodes(instance_list):
                 # attempt to re-establish the initial capacity and version
                 # check
                 logger.warning(f'Execution node attempting to rejoin as instance {hostname}.')
+                instance.health_check_started = nowtime
+                instance.save(update_fields=['health_check_started'])
                 execution_node_health_check.apply_async([hostname])
             elif instance.capacity == 0 and instance.enabled:
                 # nodes with proven connection but need remediation run health checks are reduced frequency
                 if not instance.last_health_check or (nowtime - instance.last_health_check).total_seconds() >= settings.EXECUTION_NODE_REMEDIATION_CHECKS:
                     # Periodically re-run the health check of errored nodes, in case someone fixed it
                     # TODO: perhaps decrease the frequency of these checks
                     logger.debug(f'Restarting health check for execution node {hostname} with known errors.')
+                    instance.health_check_started = nowtime
+                    instance.save(update_fields=['health_check_started'])
```

Comment:
maybe we can combine these saves with the last_seen save on line 474

Reply:
I don't believe so, no.  That save is unconditional, these need to be conditional and to be tied to the launch of the health check tasks.  We don't want to make these unconditional and then not actually trigger a health check, because then the health check endpoint will be blocked for no good reason.

Reply:
Per the discussion yesterday, I've removed writes to this field by the automated processes.